week-2/extended-project/yoda-y-sus-apis/web_api.rb

```diff
@@ -0,0 +1,208 @@
+require_relative "web_api_methods"
+
+# Get information from APIs and Webs
+#=====================================================
+characters = get_data("https://swapi.co/api/people/")
+films = get_data("https://swapi.co/api/films/")
+planets =  get_data("https://swapi.co/api/planets/")
+species = get_data("https://swapi.co/api/species/")
+initialization_scrapping
+
+
+# Filter the data that we only need
+#=====================================================
+# characters
+#
+# name    : string
+# films   : array
+# gender  : string
+# species : array
+# id      : string
+
+characters_modified = characters.map do |value|
+  {
+    name: value["name"],
+    films: value["films"],
+    gender: value["gender"],
+    species: value["species"],
+    id: value["url"]
+  }
+end
+
+# films
+#
+# episode_id : integer
+# title      : string
+# director   : string
+# release    : string
+# characters : array
+# id         : string
+#
+
+films_modified = films.map do |value|
+  {
+    episode_id: value["episode_id"],
+    title: value["title"],
+    director: value["director"],
+    release: value["release_date"],
+    characters: value["characters"],
+    id: value["url"]
+  }
+end
+
+# planets
+#
+# name      : string
+# residents : array
+# id        : string
+
+planets_modified = planets.map do |value|
+  {
+    name: value["name"],
+    residents: value["residents"],
+    id: value["url"]
+  }
+end
+
+# species
+# name : string
+# id   : string
+
+species_modified = species.map do |value|
+  {
+    name: value["name"],
+    id: value["url"]
+  }
+end
+
+#Convert data to information
+#=====================================================
+# Attach all the string content about SW characters
+# Required content: films played by each character
+#										specie which each character belongs
+
+character_final = characters_modified.map do |value|
+	films_character = value[:films]
+	# catching film's titles and number of episodes
+  new_value = films_character.map do |film_id|
+									index = films_modified.index { |film| film[:id] == film_id }
+									film_match = films_modified[index]
+                  "Episode " + film_match[:episode_id].to_s + ": " + film_match[:title]
+              end
+	# catching specie name
+  index_species = species_modified.index { |specie| specie[:id] == value[:species][0] }
+  
```

Comment:
I think your identation is not working very well, so it's hard to read

Reply:
Yes, we must improve on that, on the left space so that it's easy to read.